public/index.tsx

```diff
@@ -97,6 +101,39 @@ const viewerSettings: Partial<GlobalViewerSettings> = {
   boundingBoxColor: [255, 255, 255] as [number, number, number],
 };
 
+async function loadDataset(dataset: string, id: string) {
+  const db = new FirebaseRequest();
+
+  const datasets = await db.getAvailableDatasets();
+
+  let datasetMeta: DatasetMetaData | undefined = undefined;
+  for (const d of datasets) {
+    const innerDatasets = d.datasets!;
```

Comment:
It appears as though the type here (`DatasetMetaData`) is recursive in this `datasets` field (`datasets` has type `{ [key: string]: DatasetMetaData }`). it may be that we expect exactly one level of nesting via this property, and the author of the type didn't want to create a separate type to do that? I'm not sure exactly though, but this assertion was there before.